vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/WebClientBase.java

```diff
@@ -259,6 +259,9 @@ public WebClientBase(HttpClient client, WebClientOptions options) {
 
   @Override
   public WebClientInternal addInterceptor(Handler<HttpContext<?>> interceptor) {
+    // If a web client is constructed using another client, interceptors could get added twice.
+    // This is an attempt to ensure that doesn't happen by keeping the most recent one added.
+    interceptors.removeIf(i -> i.getClass() == interceptor.getClass());
```

Comment:
instead we should throw an exception such as IllegalStateException and let the client code know

Reply:
well it could work if client for instance double wrap the same client, better be aware of the error rather than silently try to make some magic

Reply:
Done, throws an exception now.